packages/serverpod/lib/src/server/server.dart

```diff
@@ -152,8 +157,12 @@ class Server {
     serverpod
         .logVerbose('handleRequest: ${request.method} ${request.uri.path}');
 
-    // Set Access-Control-Allow-Origin, required for Flutter web.
-    request.response.headers.add('Access-Control-Allow-Origin', '*');
+    for (var header in httpResponseHeaders.entries) {
+      // The 'Access-Control-Allow-Headers' header should be included in the
+      // response only for OPTIONS requests.
+      if (header.key == 'Access-Control-Allow-Headers') continue;
+      request.response.headers.add(header.key, header.value);
```

Comment:
This feels like a bit of black magic, and it's not clear that not all headers you pass in will be set. A better solution is to add a second parameter.

`httpOptionsResponseHeader` which can be configured just like `httpResponseHeader`.